src/tests/voteApi.test.ts

```diff
@@ -0,0 +1,106 @@
+import { test, expect } from "@playwright/test";
+import { VoteController } from "../api/controllers/voteController";
+import {
+  CreateVoteResponse,
+  PartialVoteRequest,
+} from "../api/controllers/types/voteController.type";
+import { expectToThrow } from "../utils/testUtils";
+
+test.describe("Vote API tests", () => {
+  let voteController: VoteController;
+  const body = {
+    image_id: "imageId",
+    sub_id: `my-user-${Date.now()}`,
+    value: 1,
+  };
+
+  test.beforeEach(async ({ request }) => {
+    voteController = new VoteController(request);
+  });
+
+  test("check vote up", async () => {
+    const response = await voteController.addVote(body);
+    expect(response).toMatchObject({ ...body, message: "SUCCESS" });
+  });
+
+  test("check vote down", async () => {
+    const response = await voteController.addVote({ ...body, value: -1 });
+    expect(response).toMatchObject({ ...body, value: -1, message: "SUCCESS" });
+  });
+
+  test("check that image_id is required in adding vote request", async () => {
+    const { image_id, ...bodyWithoutImage } = body;
+    await expectToThrow(
+      () => voteController.addVote(bodyWithoutImage as PartialVoteRequest),
```

Comment:
```suggestion
      () => voteController.addVote(bodyWithoutImage),
```

here and below no need to cast the type (specifying `as Type`), since the `addVote()` awaits `PartialVoteRequest` type and the `bodyWithoutImage`, `bodyWithoutValue` satisfies this type

Reply:
Fixed